File: Cpp/Source/Beggars.cpp
//
// Created by Mathi on 2022-05-23.
//

#include "Beggars.hpp"

std::vector<int> Beggars::beggars(const std::vector<int>& values, unsigned int n)
{
	auto arr = std::vector<int>(n);
	for (unsigned int i = 0; i < n; ++i)
	{
		int res = 0;
		for (unsigned int j = i; j < values.size(); j += n)
		{
			res += values[j];
		}
		arr[i] = res;
	}

	return arr;
}

File: C#/Source/Beggars.cs
// https://www.codewars.com/kata/59590976838112bfea0000fa/csharp

namespace CSharp;

public static class Beggars
{
	public static int[] Get(IReadOnlyList<int> values, int n)
	{
		var arr = new List<int>(n);
		for (var i = 0; i < n; i++)
		{
			var res = 0;
			for (int j = i; j < values.Count; j += n) res += values[j];
			arr.Add(res);
		}

		return arr.ToArray();
	}
}
